File: onyx-database-tests/src/com/onyxdevtools/embedded/entities/Person.java
package com.onyxdevtools.persist.entities;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.ManagedEntity;
import com.onyx.persistence.annotations.Attribute;
import com.onyx.persistence.annotations.Entity;
import com.onyx.persistence.annotations.Identifier;
import com.onyx.persistence.annotations.IdentifierGenerator;

/**
 * Created by tosborn1 on 10/20/15.
 */
@Entity
public class Person extends ManagedEntity implements IManagedEntity
{
    @Attribute
    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    protected long personId;

    @Attribute
    protected String firstName;

    @Attribute
    protected String lastName;

    public long getPersonId() {
        return personId;
    }

    public void setPersonId(long personId) {
        this.personId = personId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
}

File: onyx-database-tests/src/main/kotlin/com/onyxdevtools/server/entities/Person.kt
package dev.onyx.server.entities

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.ManagedEntity
import com.onyx.persistence.annotations.Attribute
import com.onyx.persistence.annotations.Entity
import com.onyx.persistence.annotations.Identifier
import com.onyx.persistence.annotations.values.IdentifierGenerator

/**
 * Created by Tim Osborn on 10/20/15.
 */
@Entity
open class Person : ManagedEntity(), IManagedEntity {

    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Suppress("unused")
    var personId: Long = 0

    @Attribute
    var firstName: String? = null

    @Attribute
    var lastName: String? = null
}
